src/modules/avformat/factory.h

```diff
@@ -0,0 +1,42 @@
+/*
```

Comment:
I like the idea of making these functions shared. But I would suggest creating a new file rather than adding on to factory. My suggestion would be to create a new file named common.h and common.c and put these new functions in there.